### ISSUE TITLE
Bugs in save and load weights

### ISSUE THREAD

**Issue by tanxchong** [fchollet/keras#2160]
I want to train a model with parts of parameters fixed. When I tried to load saved weights from the HDF5 file, it fails. It seems the saveweight function does not save none-trainable weights, while loadweight function requires them. See example code below:

<pre><code>
from keras.models import Sequential
from keras.layers.core import Merge
from keras.layers.convolutional import Convolution2D

model_range_5 = Sequential()
model_range_5.add(Convolution2D(128,5,5,dim_ordering='tf',activation='relu',border_mode='same',input_shape=(256,200,3),trainable=False))

model_range_4 = Sequential()
model_range_4.add(Convolution2D(128,5,5,dim_ordering='tf',activation='relu',border_mode='same',input_shape=(256,200,3)))

test_model = Sequential([Merge([model_range_5,model_range_4])])

test_model.compile(optimizer='SGD',loss='mse')
test_model.save_weights('weight.model',overwrite=True)
test_model.load_weights('weight.model')
</code></pre>

**Comment by SyedShayaan** [comment]
same issue, when I load the model it outputs the same value.
